File: Part 2 - Regression/Decision Tree Regression/Decision_Tree_Reg.py
#import packages
import pandas as pd
import numpy as ny
import matplotlib.pyplot as plt

#import data set
data=pd.read_csv("Position_Salaries.csv")
X=data.iloc[:, 1:2].values
y=data.iloc[:, 2].values

#model devlop
from sklearn.tree import DecisionTreeRegressor
reg=DecisionTreeRegressor(random_state=0)
reg.fit(X,y)

#model prediction
y_pred=reg.predict(6.5)

#visu
X_grid=ny.arange(min(X),max(X),0.1)
X_grid=X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color="red")
plt.plot(X_grid,reg.predict(X_grid),color='green')
plt.title("D.T.Regressor")
plt.xlabel("Position")
plt.ylabel("Salary")
plt.show()

File: Part 2 - Regression/Decision Tree Regression/Decision_Tree_Reg.R
#importing dataset
data=read.csv("Position_Salaries.csv")
data=data[2:3]



#importing req pack
install.packages("rpart")
library("rpart")

#dev model
req=rpart(formula = Salary ~ .,
          data=data,control = rpart.control(minsplit = 1))

#prediction
y_pred=predict(req,data.frame(Level=6.5))

#visualising model
library("ggplot2")
X_grid=seq(min(data$Level),max(data$Level),0.01)
ggplot()+
  geom_point(aes(x=data$Level,y=data$Salary),color="blue")+
  geom_line(aes(x=X_grid,y=predict(req,newdata = data.frame(Level=X_grid))),color="black")+
  ggtitle("Sal vs Exp using DecsionTreeRegression")+
  xlab("Exp")+
  ylab("Salary")


